tony-core/src/main/java/com/linkedin/tony/horovod/HorovodDriver.java

```diff
@@ -0,0 +1,276 @@
+/**
+ * Copyright 2021 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony.horovod;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.linkedin.tony.util.Utils;
+
+import static java.util.Objects.requireNonNull;
+
+public class HorovodDriver {
```

Comment:
Add some comments for HorovodDriver, not very intuitive to understand the role here. I know you had some content in proposal, but pls also some in the code. :)

Reply:
Got it.